### PR TITLE
[ios][expo-battery] isSupported false in simulator

### LINKED ISSUE #5584: [expo-battery] ios isSupported should return false in a simulator

### PR DESCRIPTION
**Pull request by bradbyte**
# Why

Resolves #5584.

# How

In the native iOS `EXBattery.m` module's constants method I check if the target is a simulator and then return `FALSE`. 

# Test Plan

I updated the test suite to check if `Battery.isSupported` (in addition to testing that it is `TRUE` for iOS devices and Android. It doesn't seem that the Battery test is called for simulators, and I wasn't sure if it qualified to be added to the `global.DETOX` group of tests. 

Otherwise it can be verified by checking for `Battery.isSupported` on an iOS simulator. The actual device tests for Battery should still pass since it would be `TRUE` in that case.

### CODE REVIEW

**Comment by brentvatne** [comment]
hi @bradbumbalough! we'll need to discuss this one a bit as we want to make sure that similar types of functionality has similar naming across modules. we use `Module.isAvailableAsync()` in a handful of places currently and `Module.isSupported()` in one place, but this would be the first instance of `Module.isSupported`

**Comment by bradbumbalough** [comment]
Hey @brentvatne, that makes sense! I initially just wanted to keep the existing constant but I think it would be good to have the same method of checking if it's available. We could do `Module.isSupported()` since we wouldn't need to wait on the bridge to get the value. In fact, this could probably move to pure JS function since we have the apis to know the platform and `Device.isDevice`. Let me know how you'd like it to go and I'll update.

**Comment by EvanBacon** [comment]
Could you convert it to `isAvailableAsync` -- we probably want to ensure every module has a `isAvailableAsync(): Promise<boolean>` method.

File: docs/pages/versions/v35.0.0/sdk/battery.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
 import * as Battery from 'expo-battery';
 ```
 
-Note: On iOS simulators, battery monitoring is not possible.
+Note: On iOS simulators, battery monitoring is not possible. You can confirm by checking [`Battery.isSupported`](#batteryissupported).
```
Comment (ide) [inline docs/pages/versions/v35.0.0/sdk/battery.md]: revert SDK 35 doc changes

**Review by ide — CHANGES_REQUESTED** [review]

**Pull request closed without merging**